backend/engine/plugins/cicd_tools/detectors/aws_codebuild_detector.py

```diff
@@ -33,22 +31,24 @@ def check(self, path: str) -> DetectorResult:
         }
 
         base = Path(path)
+        try:
+            buildspec_dirs = base.rglob("**/*buildspec*/")
 
-        buildspec_dirs = base.rglob("**/*buildspec*/")
+            buildspec_files = list(base.rglob("**/*buildspec*.yml"))
+            buildspec_files.extend(base.rglob("**/*buildspec*.yaml"))
 
-        buildspec_files = list(base.rglob("**/*buildspec*.yml"))
-        buildspec_files.extend(base.rglob("**/*buildspec*.yaml"))
+            for dir in buildspec_dirs:
+                yaml_files = list(dir.rglob("**/*.yml"))
+                yaml_files.extend(dir.rglob("**/*.yaml"))
 
-        for dir in buildspec_dirs:
-            yaml_files = list(dir.rglob("**/*.yml"))
-            yaml_files.extend(dir.rglob("**/*.yaml"))
+                for file in yaml_files:
+                    result["in_use"] = True
+                    result["configs"].append({"path": str(file.relative_to(path))})
 
-            for file in yaml_files:
+            for config in buildspec_files:
                 result["in_use"] = True
-                result["configs"].append({"path": str(file.relative_to(path))})
-
-        for config in buildspec_files:
-            result["in_use"] = True
-            result["configs"].append({"path": str(config.relative_to(path))})
+                result["configs"].append({"path": str(config.relative_to(path))})
 
+        except OSError as e:
+            result["errors"].append(f"Error during CodeBuild detection: {e}")
```

Comment:
Would these be better as `alerts`?

I don't think we have any guidelines for when to use one over the other. In my head, a broken symlink would also break a CICD tool that follows the link looking for a config. As a result, we likely aren't missing any valid config files when we see an error from a broken symlink and so this might be more of an `alert` than an `error`. What do you think?